src/server/routes/__tests__/catches.unit.spec.js

```diff
@@ -587,4 +590,174 @@ describe('catches.unit', () => {
+    const getCatchRequest = (payload) => ({
+      ...getServerDetails(),
+      params: { catchId: '1' },
+      payload
+    })
+
+    const getFullCatchPayload = () => ({
+      dateCaught: '2024-06-24T00:00:00+01:00',
+      species: 'species/1',
+      mass: {
+        kg: 9.61,
+        oz: 339,
+        type: 'IMPERIAL'
+      },
+      method: 'methods/1',
+      released: true,
+      onlyMonthRecorded: false,
+      noDateRecorded: false,
+      reportingExclude: false
+    })
+
+    const getFoundCatch = () => ({
+      update: jest.fn().mockResolvedValue({
+        toJSON: jest.fn().mockReturnValue({
+          id: '4',
+          dateCaught: '2024-08-02T00:00:00.000Z',
+          massKg: '2.000000',
+          massOz: '70.547924',
+          massType: 'METRIC',
+          released: false,
+          onlyMonthRecorded: true,
+          noDateRecorded: false,
+          reportingExclude: true,
+          createdAt: '2024-11-06T14:45:55.782Z',
+          updatedAt: '2025-01-03T09:24:00.032Z',
+          version: '2025-01-03T09:24:00.032Z',
+          ActivityId: '404',
+          activity_id: '404',
+          method_id: '1',
+          species_id: '1'
+        })
+      })
+    })
+
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+
+    it('should return a 200 status code if the catch is updated successfully', async () => {
+      const foundCatch = getFoundCatch()
+      Catch.findByPk.mockResolvedValueOnce(foundCatch)
+
+      const result = await patchCatchHandler(
+        getCatchRequest(getFullCatchPayload()),
+        getMockResponseToolkit()
+      )
+
+      expect(result.statusCode).toBe(200)
+    })
+
+    it('should return the updated catch if the call to update the catch is successful', async () => {
+      const foundCatch = getFoundCatch()
+      Catch.findByPk.mockResolvedValueOnce(foundCatch)
+
+      const result = await patchCatchHandler(
+        getCatchRequest(getFullCatchPayload()),
+        getMockResponseToolkit()
+      )
+
+      expect(result.payload).toMatchSnapshot()
+    })
+
+    it.each([
+      [
+        'dateCaught',
+        { dateCaught: '2024-08-02T00:00:00.000Z' },
+        { dateCaught: '2024-08-02' }
+      ],
+      ['species', { species: 'species/1' }, { species_id: '1' }],
+      [
+        'mass',
+        {
+          mass: {
+            kg: 9.61,
+            oz: 339,
+            type: 'IMPERIAL'
+          }
+        },
+        { massKg: 9.610488, massOz: 339, massType: 'IMPERIAL' }
+      ],
+      ['method', { method: 'methods/1' }, { method_id: '1' }],
+      ['released', { released: true }, { released: true }],
+      [
+        'onlyMonthRecorded',
+        { onlyMonthRecorded: true },
+        { onlyMonthRecorded: true }
+      ],
+      ['noDateRecorded', { noDateRecorded: true }, { noDateRecorded: true }],
+      [
+        'reportingExclude',
+        { reportingExclude: true },
+        { reportingExclude: true }
+      ]
```

Comment:
This might be a bit more readable in tabular format: https://jestjs.io/docs/api#2-testeachtablename-fn-timeout